engine/include/cubos/engine/render/shadows/directional_caster.hpp

```diff
@@ -19,10 +23,52 @@ namespace cubos::engine
     {
         CUBOS_REFLECT;
 
+        static constexpr int MaxSplits = 10;
+
         cubos::engine::ShadowCaster baseSettings;
 
-        int numSplits;                     ///< Number of splits for PSSM/CSM.
-        std::vector<float> splitDistances; ///< Distance of each split from 0 to 1.
-        float maxDistance = 0;             ///< Max distance (if 0, uses camera zFar).
+        int configNumSplits = 3; ///< Target number of splits for PSSM/CSM.
+        int numSplits = 0;       ///< Current number of splits for PSSM/CSM.
+
+        std::vector<float> configSplitDistances; ///< Target distance of each split's near plane from 0 to 1 (relative
+                                                 ///< to maxDistance). If the amount of distances doesn't match the
+                                                 ///< number of splits, default values will be used.
+        std::vector<float> splitDistances;       ///< Distance of each split's near plane from 0 to 1.
+
+        float maxDistance = 0;     ///< Max distance (if 0, uses camera zFar).
+        float nearDistance = 0.1F; ///< Minimum near plane distance of splits frustum.
+
+        /// @brief Directional shadow map data.
+        class ShadowMap
+        {
+        public:
+            core::gl::Texture2DArray cascades;         ///< Cascades of the shadow map.
+            core::gl::Texture2DArray previousCascades; ///< Used to check if the framebuffer needs to be recreated.
+            core::gl::Framebuffer framebuffer;         ///< Framebuffer used by the shadow rasterizer.
+
+            /// @brief Gets the size of the shadow map textures.
+            /// @return Size of the shadow map textures, in pixels.
+            glm::uvec2 getSize() const;
```

Comment:
```suggestion
            glm::uvec2 size() const;
```
For consistency with the C++ STL